### PR TITLE
Stable: standardize naming conventions

### PR DESCRIPTION
**Pull request by EndymionJkb**
We have lots of "withBpt" and "withoutBpt" names, which while helpful and self-documenting, aren't always consistent. (And are a bit awkward besides.)

Since nearly all "lower level" processing occurs with BPT removed, and generally only the outermost Vault interface functions (to and from hooks) require the full array of registered tokens, we'd like to standardize on the convention that "balances" and "index" refer to "withoutBpt" arrays. The "withBpt" arrays, as passed in, correspond to the registered tokens, so we call them registeredBalances and registeredIndexIn/Out.

scalingFactors normally do include all factors, and are "withBpt", so if we pass around scaling factors with Bpt removed, this is the exception, and we'll call them "scalingFactorsWithoutBpt".

(I tried to drop the BPT "higher" up, in `_onJoinExitPool`, since this would have been more consistent and saved a call to _dropBptItem - but was defeated by our nemesis, "stack-too-deep".)

### CODE REVIEW

File: pkg/pool-stable-phantom/contracts/StablePhantomPool.sol [hunk 1]
```diff
@@ -657,19 +688,12 @@ contract StablePhantomPool is
                     preJoinExitSupply,
                     preJoinExitInvariant,
                     currentAmp,
-                    balancesWithoutBpt,
-                    scalingFactors,
+                    balances,
+                    _dropBptItem(scalingFactors), // drop BPT to match balances
```
Comment (nventuro) [inline pkg/pool-stable-phantom/contracts/StablePhantomPool.sol:692]: I think this is unnecessary - we can do it at the place where we use the scaling factors, instead of here. And that way scaling factors will always be for the full registered set.

File: pkg/pool-stable-phantom/contracts/StablePhantomPool.sol [hunk 2]
```diff
@@ -682,19 +706,19 @@ contract StablePhantomPool is
...
+        uint256[] memory scalingFactorsWithoutBpt,
         bytes memory userData
     ) private view returns (uint256, uint256[] memory) {
         (uint256[] memory amountsIn, uint256 minBPTAmountOut) = userData.exactTokensInForBptOut();
-        InputHelpers.ensureInputLengthMatch(balancesWithoutBpt.length, amountsIn.length);
+        InputHelpers.ensureInputLengthMatch(balances.length, amountsIn.length);
 
         // The user-provided amountsIn is unscaled, so we address that.
-        _upscaleArray(amountsIn, _dropBptItem(scalingFactors));
+        _upscaleArray(amountsIn, scalingFactorsWithoutBpt);
```
Comment (nventuro) [inline pkg/pool-stable-phantom/contracts/StablePhantomPool.sol:717]: We'd then undo the change here as well.

**Review by nventuro — APPROVED** [review]

File: pkg/pool-stable-phantom/contracts/StablePhantomPool.sol (hunk 1 above)
Comment (EndymionJkb) [reply pkg/pool-stable-phantom/contracts/StablePhantomPool.sol:692]: Yeah, I tried to do it higher, where it would have made more sense.
